craft_application/app.py

```diff
@@ -114,6 +152,59 @@ def create_package(self) -> Path:
         :returns: Path to created package.
         """
 
+    @contextlib.contextmanager
+    def managed_instance(
+        self, instance_path: pathlib.PosixPath
+    ) -> Generator[Executor, None, None]:
+        """Run the application in managed mode."""
+        provider = self._manager.get_provider()
+        project_path = Path().resolve()
+        inode_number = project_path.stat().st_ino
+        instance_name = f"{self.name}-{self.project.name}-{inode_number}"
+        base_configuration = self._manager.get_configuration(
+            base=self.project.effective_base,
+            instance_name=instance_name,
+        )
+
+        emit.progress("Launching managed instance...")
+        with provider.launched_environment(
+            project_name=self.project.name,
+            project_path=project_path,
+            base_configuration=base_configuration,
+            instance_name=instance_name,
+            # craft_parts.Base doesn't currently define an alias, but it's always there.
+            build_base=base_configuration.alias.value,  # type: ignore[attr-defined]
+            allow_unstable=True,
+        ) as instance:
+            with emit.pause():
+                instance.mount(host_source=project_path, target=instance_path)
+            try:
+                yield instance
```

Comment:
hmmm correct me if I'm wrong but shouldn't this `yield` be inside the `emit.pause()`  context manager still? At least that's how it looks like to me in rockcraft.

Reply:
Good catch!